### PR TITLE
Feature/soy plant

### PR DESCRIPTION
**Pull request by RWL-Dittrich**

### CODE REVIEW

File: RealisticPlantSim/Assets/Scripts/Procedural Terrain/DefaultRandomPlacement.cs [hunk 1]
```diff
@@ -12,4 +22,9 @@ public override Vector3 randomizePosition(PlantGenerator plantGenerator)
```
Comment (Ragingram2) [inline RealisticPlantSim/Assets/Scripts/Procedural Terrain/DefaultRandomPlacement.cs:22]: Another way of doing this is:
Vector3 offset = new Vector3(randomXOffset,0,randomZOffset);
 Vector3 position = plantGenerator.transform.position+offset;

Its fine as it is. This is just more readable

Comment (Ragingram2) [reply RealisticPlantSim/Assets/Scripts/Procedural Terrain/DefaultRandomPlacement.cs:22]: also yeah I know its just a default impl, so not very important

File: RealisticPlantSim/Assets/Scripts/Procedural Terrain/PlantGenerator.cs [hunk 2]
```diff
@@ -77,17 +95,17 @@ public IEnumerator spawnPlantsInXZRange()
 
             PlantSpawnSettings spawnSettings = plantSpawnSettings[randomPlantIndex];
 
-            AbstractRandomPlacement randomPlacement;
+            AbstractPlacementStrategy placementStrategy;
 
-            if (spawnSettings.randomPlacement.GetType().IsAssignableFrom(typeof(AbstractRandomPlacement)))
+            if (spawnSettings.placementStrategy.GetClass().IsSubclassOf(typeof(AbstractPlacementStrategy)))
```
Comment (Ragingram2) [inline RealisticPlantSim/Assets/Scripts/Procedural Terrain/PlantGenerator.cs:100]: "if(placementStrategy is AbstractPlacementStrategy)" should also work here. also if placementStrategy is passed through the inspector you don't need to check whether its of the Abstract type, unity will only accept AbstractPlacementStrategy types

**Review by Ragingram2 — DISMISSED** [review]
Take a look at the comments before merging and see if you agree. I'm not gonna request changes because it works as it is.

File: RealisticPlantSim/Assets/Scripts/Procedural Terrain/PlantGenerator.cs (hunk 2 above)
Comment (RWL-Dittrich) [reply RealisticPlantSim/Assets/Scripts/Procedural Terrain/PlantGenerator.cs:100]: Your first point was valid. But the second one is not true. Unfortunately i need to use the type MonoScript to be able to drag a script into the inspector. If i use the abstract class you cant drag in a script.

Comment (GlynLeine) [reply RealisticPlantSim/Assets/Scripts/Procedural Terrain/PlantGenerator.cs]: you can make unity only accept `AbstractPlacementStrategy` scripts, but it requires editor code

Comment (RWL-Dittrich) [reply RealisticPlantSim/Assets/Scripts/Procedural Terrain/PlantGenerator.cs]: Ill take a look into that

**Review by Ragingram2 — DISMISSED** [review]

**Pull request merged**